irispy/sji.py

```diff
@@ -0,0 +1,181 @@
+'''
+
+
+def sji_fits_to_cube(filelist, start=0, stop=None, skip=None, grid=False):
+    """
+    Read SJI files and return a MapCube. Inherits
+    sunpy.instr.iris.SJI_to_cube to stitch multiple sji fits
+    files stitched into one mapcube.
+    Also sets colormap and normalization
+    Assumes hdu index of 0.
+
+
+    Parameters
+    ----------
+    filelist: `string` or `list`
+        File to read, if single file string detected, will
+        create a list of length 1.
+
+    start: `int`
+        Temporal axis index to create MapCube from
+
+    stop: `int`
+        Temporal index to stop MapCube at
+
+    skip: `int`
+        Temporal index to skip over MapCube at
+
+
+    Returns
+    -------
+    iris_cube: `sunpy.map.MapCube`
+        A map cube of the SJI sequence
+    """
+
+    if type(filelist) == str:
+        filelist = [filelist]
+    iris_cube = sunpy.map.MapCube()
+    for fname in filelist[0:]:
+        newmap = sunpy.instr.iris.SJI_to_cube(fname)
+        for frame in newmap[start:stop:skip]:
+            if frame.mean() < frame.max():
+                cmap = cm.get_cmap(frame._get_cmap_name())
+                vmax = frame.mean()+3.*frame.std()
+                frame.plot_settings['cmap'] = cmap
+                frame.plot_settings['norm'] = colors.LogNorm(1, vmax)
+                #  todo: iris_intscale
+                iris_cube.maps.append(frame)
+
+    #  todo: pointing correction(rot_hpc)
+
+    #  Option to overlay sun coordinates
+    if grid:
+        iris_cube[0].draw_grid(linestyle='--', color='orange')
+
+    return iris_cube
+
+
+def save_to_mp4(mc, outputfile, fps=60):
+    '''
+    Set up ffmpeg writer
+    Configures plot axes
+    Save an animation of the MapCube
+
+    Parameters
+    ----------
+    mc: `sunpy.map.MapCube`
+        Input mapcube
+    Outputfile: `str`
+        Path name of output file (.mp4)
+    fps: `int`
+        Desired frames per seconds (default=60)
+
+    Return
+    -----------
+    NoneType
+        mp4 to outputfile path
+    '''
+
+    plt.rcParams['animation.ffmpeg_path'] = '/usr/local/bin/ffmpeg'
+    writer = animation.FFMpegWriter(fps=fps, metadata=dict(artist='SunPy'),
+                                    bitrate=64000)
+    if not writer.isAvailable():
+        path = input('Set FFMpeg path: (ex. /usr/local/bin/ffmpeg)')
+        plt.rcParams['animation.ffmpeg_path'] = path
+    writer = animation.FFMpegWriter(fps=fps, metadata=dict(artist='SunPy'),
+                                    bitrate=64000)
+    if not writer.isAvailable():
+        print('Cannot find available FFMpeg Writer')
+        return
+    plt.rcParams['axes.facecolor'] = 'black'
+    plt.rcParams['xtick.direction'] = 'out'
+    plt.rcParams['ytick.direction'] = 'out'
+    ax = plt.axes(zorder=0)
+
+    ani = mc.plot(ax)
+    ani.save(outputfile, writer=writer)
+    print('Successfully saved to ' + outputfile)
+
+
+def dustbuster(filelist, clobber=False):
+    """
```

Comment:
I'm still unclear as to why this function operates on FITS files rather than on data itself, e.g. in a mapcube.  All the meta data is in the map.meta attribute.  In this implementation the user would then have to read in the dustbusted FITS file anyway.  Does it make more sense for this to take an sji mapcube object and return a new dustbusted sji mapcube?  The user can then write this out to a FITS file if they wanted.  Is there some other reasoning I don't see though?